code/z_adventurezones/safehouse.dm

```diff
@@ -0,0 +1,718 @@
+			id = "podbay_saferoom"
+			dir = NORTH
+
+			vertical
+				dir = EAST
+
+/obj/machinery/door_control/podbay/suspiciousdebris/
+	name = "suspicious debris"
+	id = "podbay_safehouse"
+	icon = 'icons/obj/adventurezones/safehouse.dmi'
+	icon_state = "doorctrl0"
+	desc = "Someone appears to have bolted this useless heap of junk to an asteroid."
+	layer = OBJ_LAYER
+	cooldown = 6 DECI SECONDS //To try and sync animation without redefining attack_hand.
+
+//SAFE ROOM PUZZLE
+
+/mob/living/carbon/human/dead_exec
+	name = "Jean Rockefeller" //This guy shows up as invisible in StrongDMM, not a clue why.
+	real_name = "Jean Rockefeller"
+	gender = MALE
+	interesting = "This guy seems like he's been through a lot."
+
+	initializeBioholder() //We need bioholder data intialised so we can use it elsewhere.
+		bioHolder.ownerName = name
+		bioHolder.mobAppearance.customization_first = new /datum/customization_style/moustache/vandyke
+		bioHolder.mobAppearance.customization_first_color = "#241200"
+		bioHolder.mobAppearance.customization_second = new /datum/customization_style/none
+		bioHolder.mobAppearance.customization_second_color = "#241200"
+		bioHolder.mobAppearance.customization_third = new /datum/customization_style/none
+		bioHolder.mobAppearance.customization_third_color = "#241200"
+		bioHolder.mobAppearance.e_color = "#363978"
+		bioHolder.mobAppearance.s_tone = "#FFCC99"
+		bioHolder.age = 52
+		bioHolder.bloodType = "O+"
+		bioHolder.mobAppearance.gender = "male"
+		bioHolder.mobAppearance.underwear = "none"
+		bioHolder.mobAppearance.u_color = "#FFFFFF"
+		bioHolder.Uid = bioHolder.CreateUid()
+		bioHolder.uid_hash = md5(bioHolder.Uid)
+		. = ..()
+
+	New()
+		..()
+
+		SPAWN_DBG(5 SECONDS)
+		for(var/mob/living/carbon/human/dead_exec/M)
+			qdel(M) //Jean's only here to set up the puzzle, we don't want them hanging around.
+
+obj/item/reagent_containers/iv_drip/dead_exec
+	desc = "A bag filled with someone's blood. It's labelled 'Jean Rockefeller'."
+	icon_state = "IV-blood"
+	mode = 1
+	initial_reagents = "blood"
+
+/obj/puzzle_spawn/saferoom_puzzle //Same principle as access_spawn (read: copy and paste) to help us set-up the puzzle.
+	name = "safe room bioholder spawn"
+	desc = "Sets up the prefab_saferoom puzzle bioHolder stuff depending on what it's placed on, then destroys itself."
+	icon = 'icons/effects/mapeditor.dmi'
+	icon_state = "access_spawn"
+
+	New()
+		..()
+		if(current_state > GAME_STATE_WORLD_INIT)
+			SPAWN_DBG(5 DECI SECONDS)
+				src.setup_blood_bioHolder() //Copy bioHolder data to blood
+				src.setup_bio_handscanner_access() //Copy bioHolder Uid to var in handscanner
+				qdel(src)
+
+	initialize()
+		..()
+		src.setup_blood_bioHolder()
+		src.setup_bio_handscanner_access()
+		qdel(src)
+
+	proc/setup_blood_bioHolder()
+		for(var/obj/item/reagent_containers/iv_drip/O in src.loc)
+			if(!O.reagents.has_reagent("blood"))
+				return
+			for(var/mob/living/carbon/human/dead_exec/M in world) //aka Jean
```

Comment:
Comment from Zewaka on #6041
_why not just spawn the dude inside the object with a variable then delete him_

Reply:
You were right that approach was much simpler. Updated accordingly - please see lines 362-378